### PR TITLE
Add sorted dictionaries to cnex

### PR DESCRIPTION
**Pull request by gitlarryf**

### CODE REVIEW

File: exec/cnex/cell.c [hunk 1]
```diff
@@ -554,13 +554,13 @@ Cell *cell_dictionaryIndexForWrite(Cell *c, TString *key)
         c->dictionary = wd;
     }
 
-    int64_t idx = dictionary_findIndex(c->dictionary, key);
-    if (idx == -1) {
-        idx = dictionary_addDictionaryEntry(c->dictionary, key, cell_newCell());
+    int64_t index = 0;
+    if (dictionary_findIndex(c->dictionary, key, &index) == FALSE) {
```
Comment (ghewgill) [inline exec/cnex/cell.c:558]: No need for `== FALSE` here. Use `!`.

File: exec/cnex/dictionary.c [hunk 2]
```diff
@@ -12,53 +12,64 @@
...
 
-struct tagTCell *dictionary_findDictionaryEntry(Dictionary *self, struct tagTString *key)
+struct tagTCell *dictionary_findDictionaryEntry(Dictionary *self, TString *key)
 {
-    int64_t idx;
+    int64_t idx = 0;
 
-    idx = dictionary_findIndex(self, key);
-    return idx == -1 ? NULL : self->data[idx].value;
+    if (dictionary_findIndex(self, key, &idx) == FALSE) {
```
Comment (ghewgill) [inline exec/cnex/dictionary.c:41]: No need for `== FALSE` here.

File: exec/cnex/dictionary.c [hunk 3]
```diff
@@ -12,53 +12,64 @@
...
-    int64_t idx;
-
-    idx = dictionary_findIndex(self, key);
-    if (idx != -1) {
-        cell_freeCell(self->data[idx].value);
-        self->data[idx].value = value;
-        return idx;
+    // If we weren't provided an index to add at, then we'll find out where this value should be inserted at.
+    if (index < 0) {
+        if (dictionary_findIndex(self, key, &index) == TRUE) {
```
Comment (ghewgill) [inline exec/cnex/dictionary.c:51]: Definitely no need for `== TRUE`!

File: exec/cnex/dictionary.c [hunk 4]
```diff
@@ -12,53 +12,64 @@
...
+    self->data[index].value = value;
+    return index;
 }
 
 void dictionary_removeDictionaryEntry(Dictionary *self, TString *key)
 {
-    int64_t idx = dictionary_findIndex(self, key);
-    if (idx == -1) {
+    int64_t idx = 0;
+    if (dictionary_findIndex(self, key, &idx) == FALSE) {
```
Comment (ghewgill) [inline exec/cnex/dictionary.c:72]: Again, no `== FALSE`.

File: exec/cnex/cell.c (hunk 1 above)
Comment (gitlarryf) [reply exec/cnex/cell.c:558]: Changed

File: exec/cnex/dictionary.c (hunk 2 above)
Comment (gitlarryf) [reply exec/cnex/dictionary.c:41]: Changed

File: exec/cnex/dictionary.c (hunk 3 above)
Comment (gitlarryf) [reply exec/cnex/dictionary.c:51]: Changed.

File: exec/cnex/dictionary.c (hunk 4 above)
Comment (gitlarryf) [reply exec/cnex/dictionary.c:72]: Changed.

**Pull request merged**